File: desktopclient/src/main/java/com/goexp/galgame/gui/model/Game.java
package com.goexp.galgame.gui.model;

import com.goexp.galgame.common.model.CommonGame;
import com.goexp.galgame.common.model.GameState;
import javafx.beans.property.SimpleObjectProperty;

import java.time.LocalDate;
import java.util.List;
import java.util.stream.Collectors;

public class Game extends CommonGame {
    public SimpleObjectProperty<GameState> state = new SimpleObjectProperty<GameState>();
    public Brand brand;
    public int star;


    public LocalDate getPublishDate() {
        return publishDate;
    }

    public String getName() {
        return name;
    }

    public String getSmallImg() {
        return smallImg;
    }

    public String getWebsite() {
        return website;
    }

    public int getId() {
        return id;
    }

    public String getStory() {
        return story;
    }

    public String getWriter() {
        return writer.stream().collect(Collectors.joining(","));
    }

    public String getPainter() {
        return painter.stream().collect(Collectors.joining(","));
    }

    public String getType() {
        return type.stream().collect(Collectors.joining(","));
    }

    public List<String> getTag() {
        return tag;
    }

    public String getIntro() {
        return intro;
    }

    public GameState getState() {
        return state.get();
    }

    public void setState(GameState isLike) {
        this.state.set(isLike);
    }

    public SimpleObjectProperty<GameState> stateProperty() {
        return state;
    }

    public int getStar() {
        return star;
    }

    public Brand getBrand() {
        return brand;
    }


    @Override
    public String toString() {
        return "Game{" +
                "state=" + state +
                ", brand=" + brand +
                "} " + super.toString();
    }
}

File: desktopclient/src/main/scala/com/goexp/galgame/gui/model/Game.scala
package com.goexp.galgame.gui.model

import com.goexp.common.util.string.ConsoleColors.RED
import com.goexp.common.util.string.StringOption
import com.goexp.galgame.common.model.game.{CommonGame, GameLocation, GameState}
import javafx.beans.property.{SimpleIntegerProperty, SimpleObjectProperty}

import java.util.StringJoiner
import scala.beans.BeanProperty

class Game extends CommonGame {
  @BeanProperty lazy val state = new SimpleObjectProperty[GameState]
  @BeanProperty lazy val location = new SimpleObjectProperty[GameLocation]
  @BeanProperty lazy val star = new SimpleIntegerProperty()

  @BeanProperty var brand: Brand = _

  def getWriter: String = String.join(",", writer)


  def getType: String = String.join(",", `type`)

  override def toString = s"Game[${RED.s(id.toString)}] ${RED.s(name)} Date:${publishDate} img:${smallImg}  state:<${
    Option(state).map {
      _.get
    }.getOrElse("--")
  }>"

  def infoView: String = {
    new StringJoiner(", ", classOf[Game].getSimpleName + "[", "]")
      .add("id=" + RED.s(String.valueOf(id)))
      .add("name='" + RED.s(name) + "'")
      .add("publishDate=" + publishDate)
      //      .add("writer=" + writer)
      //      .add("painter=" + painter)
      //      .add("type=" + `type`)
      //      .add("tag=" + tag)
      .add("state=" + state.get)
      .add("smallImg='" + smallImg + "'")
      //      .add("star=" + star)
      .add("\nbrand=" + brand)
      .add("\ngameImgs=" + Option(gameImgs).map(_.size).getOrElse(0))
      .add("\ngameCharacters=" + gameCharacters)
      .toString
  }

  def isOkImg = StringOption(smallImg).exists(_.startsWith("http"))
}